### PR TITLE
Helm values on inflator config plugin

### PR DESCRIPTION
**Pull request by epcim**
Proper PR for accidentally merged https://github.com/kubernetes-sigs/kustomize/pull/3316

Sbd. asked to split into packages

Let me know whether welcome and suggest updates.

* It solve issue with relative path to values + allows some merge capabilities for values.yaml
* Would be nice to rename to: Values (map[string]interface{} and today Values -> ValuesFile string as a path, but such change now would not be backward compatible.

Please discuss whether 
* ValuesLocal, ValuesMerge are good "stable" labels for what they do and we want to use them

Tests
* currently disabled, see: https://github.com/kubernetes-sigs/kustomize/pull/3206
* i am open to add some + doc once functional
* works in my use-case, but wider audience testing it, merge function, other features etc.. before it will get into an release is welcome

Depends on PR
* https://github.com/kubernetes-sigs/kustomize/pull/3352 (reason for two commits in PR)

Related:
* https://github.com/kubernetes-sigs/cli-experimental/pull/65

### CODE REVIEW

File: plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go [hunk 1]
```diff
@@ -112,6 +133,50 @@ func (p *HelmChartInflationGeneratorPlugin) Generate() (resmap.ResMap, error) {
 			return nil, err
 		}
 	}
+
+	// values
+	if len(p.ValuesLocal) > 0 {
+		fn := path.Join(p.ChartHome, p.ChartName, "kustomize-values.yaml")
```
Comment (Shell32-Natsu) [inline plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go:139]: Can we put these stuffs into a function?

**Comment by epcim** [comment]
```
go: downloading github.com/imdario/mergo v0.3.5
go: extracting github.com/imdario/mergo v0.3.5
cd api; /home/prow/go/bin/golangci-lint-kustomize \
  -c ../.golangci-kustomize.yml \
  run ./...
level=warning msg="[runner] Can't run linter unused: buildssa: analysis skipped: errors in package: [/home/prow/go/src/sigs.k8s.io/kustomize/api/builtins/HelmChartInflationGenerator.go:173:12: p.useValuesLocal undefined (type *HelmChartInflationGeneratorPlugin has no field or method useValuesLocal)]"
level=warning msg="[runner] Can't run linter goanalysis_metalinter: assign: failed prerequisites: inspect@sigs.k8s.io/kustomize/api/builtins"
level=error msg="Running error: assign: failed prerequisites: inspect@sigs.k8s.io/kustomize/api/builtins"
```

have to commit as with types pkg quite sure, what make target generates the api api/builtins/HelmChartInflationGenerator.go ?

**Comment by Shell32-Natsu** [comment]
I think the function name `useValuesLocal` is wrong.

**Comment by epcim** [comment]
k, now fails on 
```
"cmd/config" "api/" "kustomize/" "kyaml/"
Spanning changesets detected in the following commits:

	* 747ec5d06d24188a1e95c9134ba9cc2df8ab0cea

Commits that include multiple Go modules must be split into multiple commits that each touch no more than one module.
```

which is expected and will get solved after #3352 is merged. Right?

**Comment by Shell32-Natsu** [comment]
You only need to split your commits, not PR. Make sure every commit is only in one module.

**Pull request merged**